server/public/model/config.go

```diff
@@ -886,6 +889,14 @@ func (s *ServiceSettings) SetDefaults(isUpdate bool) {
 		s.SelfHostedPurchase = NewBool(true)
 	}
 
+	if s.UniqueEmojiReactionLimitPerPost == nil {
+		s.UniqueEmojiReactionLimitPerPost = NewInt(ServiceSettingsDefaultUniqueReactionsPerPost)
+	}
+
+	if *s.UniqueEmojiReactionLimitPerPost > ServiceSettingsMaxUniqueReactionsPerPost {
+		s.UniqueEmojiReactionLimitPerPost = NewInt(ServiceSettingsMaxUniqueReactionsPerPost)
+	}
```

Comment:
I can see many different use cases here and we want to be able to cover all of them:
1. Administrator doesn't care how many emojis they want, but it's pretty unlikely they'll ever use more than 500. To avoid crashing mobile or desktop apps, we limit it up to 500 so that they can't shoot themselves in the foot.
2. Administrator doesn't care either way how many emojis, so we once again avoid some bad UX by having too many emojis displayed at once, so 50 is a reasonable setting for that.
3. Administrator DOES care how many emojis they want, so they set a limit that gives them the acceptable UX they want.

To simplify:
- Upper limit of 500 to avoid performance/really bad UX.
- Default limit of 50 to show the ideal UX
- Customizable in case they want to change it.

This was all discussed here: https://community.mattermost.com/core/pl/f6kj98y88tgxjm66i8hrbd4fdc

Reply:
I respect the back and forth that arrived here, but it feels like a compromise that adds permanent overhead. Either we "support" 500 emojis at once, or we don't -- in which case it seems like we don't need to add complications to allow the admin to further reduce that limit. I'm advocating for reducing long term maintenance and configuration overhead.

Reply:
Following this up in the above thread.

Reply:
Deferring to the linked thread as noted earlier -- thanks again :)